sklearn/feature_extraction/_dict_vectorizer.py

```diff
@@ -98,6 +121,32 @@ def __init__(self, *, dtype=np.float64, separator="=", sparse=True,
         self.sparse = sparse
         self.sort = sort
 
+    def _add_element(self, f, v, feature_names, vocab, fitting=True,
+                     transforming=False, indices=None, values=None):
+        if isinstance(v, str):
+            feature_name = "%s%s%s" % (f, self.separator, v)
+            v = 1
+        elif isinstance(v, Number) or (v is None):
+            feature_name = f
+        elif isinstance(v, Iterable) and not isinstance(v, str):
+            for vv in v:
+                self._add_element(f, vv, feature_names, vocab,
+                                  fitting, transforming, indices, values)
```

Comment:
This means we support any level of nesting, such as:

```py
    D_in = [{"version": ["1", ["2", "1"]], "ham": 2}]
```

where the `["2", "1"]` are nested in the list. I think I would to be more restrictive and not support this.

Reply:
I _think_ it's fixed. :)